misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormatterUtils.java

```diff
@@ -0,0 +1,230 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ballerinalang.formatter.core;
+
+import io.ballerina.tools.text.LinePosition;
+import io.ballerina.tools.text.LineRange;
+import io.ballerinalang.compiler.syntax.tree.AbstractNodeFactory;
+import io.ballerinalang.compiler.syntax.tree.Minutiae;
+import io.ballerinalang.compiler.syntax.tree.MinutiaeList;
+import io.ballerinalang.compiler.syntax.tree.Node;
+import io.ballerinalang.compiler.syntax.tree.SyntaxKind;
+import io.ballerinalang.compiler.syntax.tree.Token;
+import org.wso2.ballerinalang.compiler.util.diagnotic.DiagnosticPos;
+
+import static io.ballerinalang.compiler.syntax.tree.AbstractNodeFactory.createMinutiaeList;
+import static io.ballerinalang.compiler.syntax.tree.AbstractNodeFactory.createWhitespaceMinutiae;
+
+/**
+ * Class that contains the util functions used by the formatting tree modifier.
+ */
+class FormatterUtils {
+
+    /**
+     * Get the node position.
+     *
+     * @param node node
+     * @return node position
+     */
+    static DiagnosticPos getPosition(Node node) {
+        if (node == null) {
+            return null;
+        }
+        LineRange range = node.lineRange();
+        LinePosition startPos = range.startLine();
+        LinePosition endPos = range.endLine();
+        int startOffset = startPos.offset();
+        if (node.kind().equals(SyntaxKind.FUNCTION_DEFINITION) || node.kind().equals(SyntaxKind.TYPE_DEFINITION) ||
+                node.kind().equals(SyntaxKind.CONST_DECLARATION)) {
+            startOffset = (startOffset / 4) * 4;
+        }
+        return new DiagnosticPos(null, startPos.line() + 1, endPos.line() + 1,
+                startOffset, endPos.offset());
+    }
+
+    // TODO: Use a generic way to get the parent node using querying.
+    static <T extends Node> Node getParent(T node, SyntaxKind syntaxKind) {
+        Node parent = node.parent();
+        if (parent == null) {
+            parent = node;
+        }
+        SyntaxKind parentKind = parent.kind();
+        if (parentKind == SyntaxKind.MODULE_VAR_DECL) {
+            if (parent.parent() != null && parent.parent().kind() == SyntaxKind.MODULE_PART &&
+                    syntaxKind == SyntaxKind.QUALIFIED_NAME_REFERENCE) {
+                return null;
+            }
+            return parent;
+        } else if (parentKind == SyntaxKind.FUNCTION_DEFINITION ||
+                parentKind == SyntaxKind.IF_ELSE_STATEMENT ||
+                parentKind == SyntaxKind.ELSE_BLOCK ||
+                parentKind == SyntaxKind.WHILE_STATEMENT ||
+                parentKind == SyntaxKind.CONST_DECLARATION ||
+                parentKind == SyntaxKind.METHOD_DECLARATION ||
+                parentKind == SyntaxKind.TYPE_DEFINITION) {
+            return parent;
+        } else if (syntaxKind == SyntaxKind.SIMPLE_NAME_REFERENCE) {
```

Comment:
We could use a switch-case in-place of the if-else. It can nicely handle these multiple matches as well.

Reply:
Planning on refactoring and simplifying the logic separately in a different PR.